src/node.rs

```diff
@@ -48,7 +49,11 @@ impl Node {
     }
 
     pub fn increment_age(&mut self) {
-        self.age = self.age.checked_add(1).expect("age overflow")
+        if self.age == u8::MAX - 1 {
+            error!("Node {:?} reached maximum age {}", self.name, self.age + 1);
```

Comment:
not sure, but I thought we are more interested in the overall network status instead of particular node when some node hits the age boundary?

Reply:
There is also a complete age distribution printed out at the end of the simulation (after pressing Ctrl+C). That should give us enough info, shouldn't it?